src/components/individual-transcript/IndividualTranscript.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+import React, { useState } from "react";
+import { Transcript } from "contentlayer/generated";
+import BreadCrumbs from "../common/BreadCrumbs";
+import TranscriptMetadataComponent from "../common/TranscriptMetadataCard";
+import ContentSwitch from "../common/ContentSwitch";
+import ContentGrouping from "../explore/ContentGrouping";
+import { createSlug, extractListOfHeadings } from "@/utils";
+import Wrapper from "../layout/Wrapper";
+import { twMerge } from "tailwind-merge";
+
+const IndividualTranscript = ({ transcript }: { transcript: Transcript }) => {
+  const [currentHeading, setCurrentHeading] = useState("");
+ 
+  const allHeadings = extractListOfHeadings(transcript.body.raw).map(
+    (heading) => {
+      return {
+      name: heading.replace(/[#]+\s+/gi, ""),
+      slug: createSlug(heading),
+      count: 0,
+      }
+    }
+  );
+  
+  const groupedHeading = allHeadings.reduce(
+    (heading, headingNow) => ({ ...heading, [headingNow.name]: headingNow }),
+    {}
+  );
```

Comment:
what is this doing?

Reply:
So this turns the array to an object that has name, slug (for the href). We need this because of how the component `ContentGrouping` was structured.

`{ 
Audience Questions: {name: 'Audience Questions', slug: 'audience-questions', count: 0},
How should companies approach developers?: {name:'', slug:'',count:0} 
}
`

Reply:
Okay sure, 
first, we have a component called `ContentGrouping` now this component only collects a type of `Record<string, ContentData[]>`; 

The reason for this is because of how our JSONs (tag-data and types-data.json) are structured.

Now since this is what `ContentGrouping` component accepts we have to structure our data in that format.

So we could agree that this could be done earlier (at the point where we map the heading), which is what I will explore.

Reply:
Thank you for pointing this out, @kouloumos. I see your concerns about the  "unnecessary complexity" in the type definitions and simplifying the `ContentGrouping` component. You raise a good point.

Would you prefer that I address these issues in this PR, or should we open a new issue to tackle them separately? If it's the latter, I would appreciate if could create an issue tackling this and the other comments on this PR. Let me know your preference.

I really appreciate the constructive feedback.

Reply:
Thank you for your response! I don’t see this as a blocker, so I’m fine with moving forward and merging this for now.

That said, I saw this as an opportunity to highlight a broader issue, which is why I brought it up. I believe it’s something we should finally take the time to address. This isn’t meant to reflect on you specifically but rather to open up a team-wide discussion (cc @Emmanuel-Develops ) about our processes and culture as a whole.